### PR TITLE
Refactor API to use Provider pattern.

### PR DESCRIPTION
**Pull request by dgoodwin**
Preparation for moving to the actuator pattern used by upstream
cluster-api. Will keep the AWS types here for now, but we can split them
out later.

### CODE REVIEW

**Comment by dgoodwin** [comment]
Next phase will add the actual actuators, this is just the API moving to providers.

File: config/crds/credminter_v1beta1_awsproviderspec.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+          type: array
+      required:
+      - statementEntries
+  version: v1beta1
+status:
+  acceptedNames:
+    kind: ""
+    plural: ""
+  conditions: []
+  storedVersions: []
```
Comment (dgoodwin) [inline config/crds/credminter_v1beta1_awsproviderspec.yaml]: Hmm, not great that this results in crd's getting created. I could remove them in the make target after generation, but maybe there's a way to disable CRD generation but still generate the runtime object code.

Comment (csrwng) [reply config/crds/credminter_v1beta1_awsproviderspec.yaml]: You can avoid crd generation if you omit ObjectMeta from the type ... it shouldn't be needed

Comment (dgoodwin) [reply config/crds/credminter_v1beta1_awsproviderspec.yaml]: Aha thanks, updated.

**Comment by dgoodwin** [comment]
Need a review here @csrwng or @joelddiaz

**Comment by csrwng** [comment]
@dgoodwin were you thinking of introducing an actuator interface in later PR?

**Comment by dgoodwin** [comment]
Yes I have that ready to go I think, it's huge though. Happy to push it all at once if you'd prefer.

**Comment by csrwng** [comment]
No, that’s fine. With PR’s small > big :-)

**Pull request merged**